From 0789a6232bc3e18836cd241618ce8acda15843ca Mon Sep 17 00:00:00 2001
From: Gurkengewuerz <niklas@schuetrumpf.net>
Date: Sat, 16 Sep 2023 21:43:11 +0200
Subject: [PATCH] feat: use .usb to detect usb drivers

---
 tools/run_copy_st_drivers.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tools/run_copy_st_drivers.py b/tools/run_copy_st_drivers.py
index b39968f..167010d 100755
--- a/tools/run_copy_st_drivers.py
+++ b/tools/run_copy_st_drivers.py
@@ -313,6 +313,17 @@ if '-clean' in sys.argv: clean = True
 #do_for_each_folder(clean=True)
 do_for_each_target(clean=clean,silent=silent)
 
+#################################################################
+dirlist = os.listdir(mLRSdirectory)
+for f in dirlist:
+    if f[:3] == "rx-" or f[:3] == "tx-":
+        useUSB = os.path.join(mLRSdirectory,f,'.usb')
+        if os.path.isfile(useUSB):
+            print('copy usb drivers for', f)
+            copy_usb_device_library_driver(f,clean=clean,silent=silent)
+sys.exit(0)
+#################################################################
+
 copy_usb_device_library_driver('tx-FRM303-f072cb', clean=clean,silent=silent)
 
 print('# DONE #')
-- 
2.39.2.windows.1

