From 8ed8c4764f1adf505fc1899b5c76094514cee7d0 Mon Sep 17 00:00:00 2001
From: Gurkengewuerz <niklas@schuetrumpf.net>
Date: Sat, 16 Sep 2023 18:24:16 +0200
Subject: [PATCH] fix: added missing HAL Driver for headless build

---
 tools/run_make_firmwares3.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/run_make_firmwares3.py b/tools/run_make_firmwares3.py
index 71413cc..63782bb 100755
--- a/tools/run_make_firmwares3.py
+++ b/tools/run_make_firmwares3.py
@@ -223,6 +223,8 @@ MLRS_SOURCES_HAL_STM32G4 = [
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_hal_pwr_ex.c'),
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_hal_rcc.c'),
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_hal_rcc_ex.c'),
+    os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_hal_pcd.c'),
+    os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_hal_pcd_ex.c'),
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_ll_adc.c'),
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_ll_comp.c'),
     os.path.join('Drivers','STM32G4xx_HAL_Driver','Src','stm32g4xx_ll_cordic.c'),
-- 
2.42.0.windows.2

